preflight_check/check.py

```diff
@@ -0,0 +1,254 @@
+    """
+    task_inst = task.TaskCheck("Checking STONITH/Fence")
+    with task_inst.run():
+        if not utils.FenceInfo().fence_enabled:
+            task_inst.warn("stonith is disabled")
+            return
+
+        task_inst.info("stonith is enabled")
+        rc, outp, _ = crmshutils.get_stdout_stderr("crm_mon -r1 | grep '(stonith:.*):'")
+        if rc != 0:
+            task_inst.warn("No stonith resource configured!")
+            return
+
+        res = re.search(r'([^\s]+)\s+\(stonith:(.*)\):\s+(\w+)', outp)
+        res_name, res_agent, res_state = res.groups()
+        common_msg = "stonith resource {}({})".format(res_name, res_agent)
+        state_msg = "{} is {}".format(common_msg, res_state)
+
+        task_inst.info("{} is configured".format(common_msg))
+        if res_state == "Started":
+            task_inst.info(state_msg)
+        else:
+            task_inst.warn(state_msg)
+
+        if re.search(r'sbd$', res_agent):
+            if crmshutils.service_is_active("sbd"):
+                task_inst.info("sbd service is running")
+            else:
+                task_inst.warn("sbd service is not running!")
+
+
+def check_nodes():
+    """
+    Check nodes info:
+      Current DC
+      Quorum status
+      Online/OFFLINE/UNCLEAN nodes
+    """
+    task_inst = task.TaskCheck("Checking nodes")
+    with task_inst.run():
+        rc, outp, errp = crmshutils.get_stdout_stderr("crm_mon -1")
+        if rc == 0:
+            # check DC
+            res = re.search(r'Current DC: (.*) \(', outp)
+            if res:
+                task_inst.info("DC node: {}".format(res.group(1)))
+
+            # check quorum
+            if re.search(r'partition with quorum', outp):
+                task_inst.info("Cluster have quorum")
+            else:
+                task_inst.warn("Cluster lost quorum!")
+
+            # check Online nodes
+            res = re.search(r'Online:\s+(\[.*\])', outp)
+            if res:
+                task_inst.info("Online nodes: {}".format(res.group(1)))
+
+            # check OFFLINE nodes
+            res = re.search(r'OFFLINE:\s+(\[.*\])', outp)
+            if res:
+                task_inst.warn("OFFLINE nodes: {}".format(res.group(1)))
+
+            # check UNCLEAN nodes
+            res = re.findall(r'Node (.*): UNCLEAN', outp)
+            for item in res:
+                task_inst.warn('Node {} is UNCLEAN!'.format(item))
+        else:
+            task_inst.error("run \"crm_mon -1\" error: {}".format(errp))
+
+
+def check_resources():
+    """
+    Check number of Started/Stopped/FAILED resources
+    """
+    task_inst = task.TaskCheck("Checking resources")
+    with task_inst.run():
+        started_list = []
+        stopped_list = []
+        failed_list = []
+        cmd = "crm_mon -r1"
+
+        rc, outp, errp = crmshutils.get_stdout_stderr(cmd)
+        if rc == 0:
+            res = re.findall(r'([^\s]+)\s+\(.*\):\s+(\w+) ', outp)
```

Comment:
This won't show the clone resource, for example:
```
  * libvirt_stonith	(stonith:external/libvirt):	 Started SLE15-sp2-gm-nwang-crmshreview-node1
  * Clone Set: ms_1-multi-0 [res-1-multi-0] (promotable):
    * Masters: [ SLE15-sp2-gm-nwang-crmshreview-node2 ]
    * Slaves: [ SLE15-sp2-gm-nwang-crmshreview-node1 ]
```

Reply:
Changed.
Instead of using un-readable regexp, here just calling `crmsh.completers.resources_started` and `crmsh.completers.resources_stopped` to fetch the resources list